server/policies/share.js

```diff
@@ -5,14 +5,13 @@ import policy from "./policy";
 
 const { allow } = policy;
 
-allow(
-  User,
-  ["read", "update"],
-  Share,
-  (user, share) => user.teamId === share.teamId
-);
+allow(User, ["read", "update"], Share, (user, share) => {
+  if (user.isViewer) return false;
```

Comment:
I think viewers can read share information, that seems expected or they won't be able to see if a doc is shared publicly and access that link

Reply:
Done